### ISSUE TITLE
How exactly is the cropping done?

### ISSUE THREAD

**Issue by lagidigu** [luckycallor/InsightFace-tensorflow#10]
Hello,

Thanks for making this repo!
Just one question: Is cropping the image with MTCNN sufficient when using the ms1m dataset? Were the images aligned with a margin of 5?

Thanks a lot in advance!

**Comment by luckycallor** [comment]
@lagidigu 
When using the dataset opensourced by the official InsightFace project, there is no need to do further cropping.

**Comment by lagidigu** [comment]
@luckycallor 
Thanks for the quick answer. I'm trying to generate embeddings with the pre-trained network (ms1m-100) for later inference with nearest neighbour.

I am asking because it seems like inference with cosine distance is not precise. Cropping is consistent with the original dataset cropping, so there shouldn't be a problem. 

Could it be due to the batch size being static?

**Comment by Wxsang** [comment]
@lagidigu Actually, i had the same problem. In my case, i need to identify someone. I found the margin of MTCNN have a big impact on the final result. if the value of margin too small, the accuracy goes down. i dont konw why.

**Comment by lagidigu** [comment]
@Wxsang I currently use a margin of 5, so that could explain it. What margin have you found the most success with?

**Comment by Wxsang** [comment]
@lagidigu
I am sorry to reply you so late. I can't find the best margin, all results are not good.
